### ISSUE TITLE
[BUG]matplotlib 3.9 broken dependency

### ISSUE THREAD

**Issue by ljw20180420** [open2c/coolpuppy#160]
As in https://github.com/open2c/cooltools/pull/533, please also fix this for coolpuppy.

**Comment by Phlya** [comment]
I think if you upgrade cooltools it should fix coolpuppy, doesn't it?

**Comment by ljw20180420** [comment]
No. My cooltools are the newest version 0.7.1. My coolpuppy is the newest version 1.1.0. However, the error persists.
```bash
/home/ljw/wuqiang/sx/sx_lcy/.conda/lib/python3.11/site-packages/coolpuppy/coolpup.py:2224: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  pups["expected"] = pups["expected"].fillna(False)
/home/ljw/wuqiang/sx/sx_lcy/.conda/lib/python3.11/site-packages/coolpuppy/lib/io.py:53: FutureWarning: Starting with pandas version 3.0 all arguments of to_hdf except for the argument 'path_or_buf' will be keyword-only.
  ].to_hdf(filename, "annotation", mode=mode)
INFO:coolpuppy:Saved output to www/5beae7ee9f513a263af4dc494c4c9b54/file2e9e059726cdc.1.clpy
Traceback (most recent call last):
  File "/home/ljw/wuqiang/sx/sx_lcy/.conda/lib/python3.11/site-packages/cooltools/lib/plotting.py", line 6, in <module>
    from matplotlib.cm import register_cmap
ImportError: cannot import name 'register_cmap' from 'matplotlib.cm' (/home/ljw/wuqiang/sx/sx_lcy/.conda/lib/python3.11/site-packages/matplotlib/cm.py)
```
I use the command
```bash
plotpup.py --input_pups file.clpy -o file.pdf
```

**Comment by Phlya** [comment]
As you can see, the error comes from cooltools. coopuppy doesn't run this command directly.

**Comment by ljw20180420** [comment]
Yes! You are right. So cooltools does not fix the bug complete in 0.7.1? I will close this and open a new one in cooltools.
